crates/fuel-core/src/graphql_api/storage.rs

```diff
@@ -249,4 +249,53 @@ where
         self.commit()?;
         Ok(())
     }
+
+    #[cfg(feature = "fault-proving")]
+    fn registry_root(&self) -> StorageResult<Bytes32> {
+        use fuel_core_storage::tables::merkle::DenseMetadataKey;
+
+        use crate::graphql_api::storage::da_compression::v2::{
+            address::TemporalRegistryAddressMerkleMetadata as AddressMerkle,
+            asset_id::TemporalRegistryAssetIdMerkleMetadata as AssetIdMerkle,
+            contract_id::TemporalRegistryContractIdMerkleMetadata as ContractIdMerkle,
+            evictor_cache::TemporalRegistryEvictorCacheMerkleMetadata as EvictorCacheMerkle,
+            predicate_code::TemporalRegistryPredicateCodeMerkleMetadata as PredicateCodeMerkle,
+            registry_index::TemporalRegistryIndexMerkleMetadata as RegistryIndexMerkle,
+            script_code::TemporalRegistryScriptCodeMerkleMetadata as ScriptCodeMerkle,
+            timestamps::TemporalRegistryTimestampsMerkleMetadata as TimestampsMerkle,
+        };
+
+        macro_rules! get_merkle_root {
+            ($self:ident, $metadata:ty) => {
+                $self
+                    .storage::<$metadata>()
+                    .get(&DenseMetadataKey::Latest)?
+                    .map(|cow| *cow.root())
+                    // the only time it will be empty is on initialization
+                    .unwrap_or(*Bytes32::default())
```

Comment:
This should be `unwrap_or_default`. But I'd prefer to return an error if this isn't set to prevent silent bugs from causing the chain to halt.

Reply:
Or is this the intended behavior?

Reply:
addressed in d6f0ed9869